catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1522,6 +1525,85 @@ public UpdateResponse update(UpdateRequest updateRequest)
         return updateResponse;
     }
 
+    private DeleteRequest rewriteQuery(DeleteRequest deleteRequest)
+            throws UnsupportedQueryException {
+        if (Metacard.ID.equals(deleteRequest.getAttributeName())) {
+            return deleteRequest;
+        }
+
+        Filter filter = frameworkProperties.getFilterBuilder()
+                .anyOf(deleteRequest.getAttributeValues()
+                        .stream()
+                        .map(e -> frameworkProperties.getFilterBuilder()
+                                .attribute(deleteRequest.getAttributeName())
+                                .is()
+                                .equalTo()
+                                .text((String) e))
+                        .collect(Collectors.toList()));
+        SourceResponse response = catalog.query(new QueryRequestImpl(new QueryImpl(filter), false));
+        List<Entry<Serializable, Metacard>> newUpdateList = response.getResults()
+                .stream()
+                .filter(r -> MetacardVersion.isNotVersion(r.getMetacard()))
+                .map(r -> new AbstractMap.SimpleEntry<>((Serializable) r.getMetacard()
+                        .getId(), r.getMetacard()))
+                .collect(Collectors.toList());
+
+        // confirm every item from original update list is present.
+        newUpdateList.stream()
```

Comment:
You're not doing anything with the boolean result of this. Unless I'm missing where a RuntimeException is being thrown, this is a noop.